src/server.cpp

```diff
@@ -20,19 +22,50 @@ void Server::start_receive()
         std::bind(&Server::handle_receive, this, pls::_1, pls::_2));
 }
 
-void Server::handle_receive(const error_code& error, std::size_t /*bytes_transferred*/)
+void Server::handle_receive(const error_code& error, std::size_t bytes)
 {
     if (!error || error == boost::asio::error::message_size)
     {
-        const std::string message("Hello, World!");
+        size_t length = recv_buffer_[2] * 256 + recv_buffer_[3];
```

Comment:
What if we receive less than 4 bytes? `recv_buffer_[2]` and `recv_buffer_[3]` would be undefined.